packages/story-editor/src/components/panels/document/taxonomies/FlatTermSelector.js

```diff
@@ -129,6 +157,37 @@ function FlatTermSelector({ taxonomy, canCreateTerms }) {
         <Tags.Description id={`${taxonomy.slug}-description`}>
           {taxonomy.labels.separate_items_with_commas}
         </Tags.Description>
+        {mostUsed.length > 0 && (
+          <WordCloud.Wrapper>
+            <WordCloud.Heading>
+              {__('Most Used', 'web-stories')}
+            </WordCloud.Heading>
+            <WordCloud.List>
+              {mostUsed.map((term, i) => (
+                <WordCloud.ListItem key={term.id}>
+                  <WordCloud.Word
+                    onClick={() => {
+                      if (terms[taxonomy.restBase]?.includes(term.id)) {
+                        return;
+                      }
+                      addTermToSelection(taxonomy, term);
+                    }}
+                  >
+                    {term.name}
+
+                    {i < mostUsed.length - 1 &&
+                      /* translators: delimiter used in a list */
+                      __(',', 'web-stories')}
+                  </WordCloud.Word>
+                  {
+                    /* Browser only respects the white space in the li, not the button */
+                    i < mostUsed.length - 1 && ' '
+                  }
```

Comment:
💭 Would the il8n utils for lists be helpful here?
https://github.com/google/web-stories-wp/blob/main/packages/i18n/src/translateToInclusiveList.js
https://github.com/google/web-stories-wp/blob/main/packages/i18n/src/translateToExclusiveList.js

Reply:
I reached for those originally, but don't they include `and` or `or` in the list? the list here is just supposed to be a comma separated list where each item is a button.

Is there a way to use those functions that wouldn't be more trouble than it's worth to augment them for this use case?

Reply:
Ah then yeah they wouldn't make a lot of sense. Makes sense to just combine using a separator. We could always make a util for that too that doesn't use any conjunctions 🤔

Reply:
yea I guess my only concern is that it's not a list of words, but instead a list of components each with different props. Only worry there is the abstraction being more complexity when we're only running into this use case here.

Do your utils above handle react nodes like this? or just join words and return a string?

Reply:
I think they just join words, so it would be more complex to render react nodes. Seems like a util is a problem for another day